File: 0056-merge-intervals/0056-merge-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    intervals.sort((a,b)=>a[0]-b[0]);
    let res=[];
    let s1,e1,s2,e2;
    s1=intervals[0][0];
    e1=intervals[0][1];
    for(let i=1;i<intervals.length;i++){
        s2=intervals[i][0];
        e2=intervals[i][1];
        if(e1>=s2){
            s1=s1;
            e1=Math.max(e1,e2);
            continue;
        }
        res.push([s1,e1]);
        s1=s2;
        e1=e2;
    }
    res.push([s1,e1]);
    return res;
};

File: 0056-merge-intervals/0056-merge-intervals.java
import java.util.*;

class Solution {
    public int[][] merge(int[][] intervals) {

        Arrays.sort(intervals, (a, b) -> a[0] - b[0]);

        ArrayList<int[]> arr = new ArrayList<>();

        int s1 = intervals[0][0];
        int e1 = intervals[0][1];

        for (int i = 1; i < intervals.length; i++) {

            int s2 = intervals[i][0];
            int e2 = intervals[i][1];

            if (e1 >= s2) {
                e1 = Math.max(e1, e2);
            } 
            else {
                arr.add(new int[]{s1, e1});

                s1 = s2;
                e1 = e2;
            }
        }

        // Add the last interval
        arr.add(new int[]{s1, e1});

        return arr.toArray(new int[arr.size()][]);
    }
}